component/attr_cache/cacheMap_test.go

```diff
@@ -0,0 +1,377 @@
+	}
+
+}
+
+func (suite *cacheMapTestSuite) TestInsertFileCacheMap() {
+	//create path string in form of test/dir/file
+	path := "a/c1/TestFile.txt"
+	startTime := time.Now()
+	attr := internal.CreateObjAttr(path, 1024, startTime)
+
+	//insert path into suite.rootAttrCacheItem
+	suite.rootAttrCacheItem.insert(attr, true, startTime)
+
+	//verify correct values are in cacheMapTree
+	cachedItem, err := suite.rootAttrCacheItem.get(path)
+	suite.assert.Nil(err)
+	suite.assert.NotNil(cachedItem)
+	suite.assert.EqualValues(path, cachedItem.attr.Path)
+	suite.assert.EqualValues(1024, cachedItem.attr.Size)
+	suite.assert.EqualValues(false, cachedItem.attr.IsDir())
+}
+
+func (suite *cacheMapTestSuite) TestInsertFolderCacheMap() {
+	//create path string in form of test/dir/file
+	path := "a/c1/TestFolder"
+	startTime := time.Now()
+	attr := internal.CreateObjAttrDir(path)
+
+	//insert path into suite.rootAttrCacheItem
+
+	suite.rootAttrCacheItem.insert(attr, true, startTime)
+
+	//verify correct values are in cacheMapTree
+	cachedItem, err := suite.rootAttrCacheItem.get(path)
+	suite.assert.Nil(err)
+	suite.assert.NotNil(cachedItem)
+	suite.assert.EqualValues(path, cachedItem.attr.Path)
+	suite.assert.EqualValues(4096, cachedItem.attr.Size)
+	suite.assert.EqualValues(true, cachedItem.attr.IsDir())
+}
+
+func (suite *cacheMapTestSuite) TestInsertDirsAndFiles() {
+	path := "a/c1/c12/c3p0/newfile.txt"
+	startTime := time.Now()
+	attr := internal.CreateObjAttr(path, 1024, startTime)
+
+	suite.rootAttrCacheItem.insert(attr, true, startTime)
+
+	cachedItem, err := suite.rootAttrCacheItem.get(path)
+
+	suite.assert.Nil(err)
+	suite.assert.NotNil(cachedItem)
+	suite.assert.EqualValues(path, cachedItem.attr.Path)
+	suite.assert.EqualValues(1024, cachedItem.attr.Size)
+	suite.assert.EqualValues(false, cachedItem.attr.IsDir())
+}
+
+func (suite *cacheMapTestSuite) TestMarkDeleted() {
+	deleteTime := time.Now()
+
+	//insert an item
+	path := "a/c1/TempFile.txt"
+	startTime := time.Now()
+	attr := internal.CreateObjAttr(path, 1024, startTime)
+
+	//insert path into suite.rootAttrCacheItem
+	suite.rootAttrCacheItem.insert(attr, true, startTime)
+
+	//validate it is there
+	cachedItem, err := suite.rootAttrCacheItem.get(path)
+	suite.assert.Nil(err)
+	suite.assert.NotNil(cachedItem)
+	suite.assert.EqualValues(path, cachedItem.attr.Path)
+	suite.assert.EqualValues(1024, cachedItem.attr.Size)
+	suite.assert.EqualValues(startTime, cachedItem.attr.Mtime)
+	suite.assert.EqualValues(false, cachedItem.attr.IsDir())
+	suite.assert.EqualValues("TempFile.txt", cachedItem.attr.Name)
+	suite.assert.EqualValues(true, cachedItem.attrFlag.IsSet(AttrFlagValid))
+
+	// mark it deleted
+	cachedItem.markDeleted(deleteTime)
+
+	//verify it is marked deleted
+	cachedItem, err = suite.rootAttrCacheItem.get(path)
+	suite.assert.Nil(err)
+	suite.assert.NotNil(cachedItem)
+	suite.assert.EqualValues("", cachedItem.attr.Path)
+	suite.assert.EqualValues(true, cachedItem.isDeleted())
```

Comment:
Whether the entry exists is checked three different times in this block of asserts.
Maybe there could be a separate test that verifies that `isDeleted()` works, and one that verifies that `exists()` works properly. And then everywhere else, you can just test the flag once.